integration_tests/rebalance_test.go

```diff
@@ -0,0 +1,174 @@
+			Vote: &types.RebalanceVote{
+				Cellar: &types.Cellar{
+					Id: hardhatCellar.String(),
+					TickRanges: []*types.TickRange{
+						{Upper: 198840, Lower: 192180, Weight: 100},
+					},
+				},
+				CurrentPrice: 100,
+			},
+			Salt: "testsalt",
+		}
+
+		s.T().Logf("checking that test cellar exists in the chain")
+		val := s.chain.validators[0]
+		s.Require().Eventuallyf(func() bool {
+			kb, err := val.keyring()
+			s.Require().NoError(err)
+			clientCtx, err := s.chain.clientContext("tcp://localhost:26657", &kb, "val", val.keyInfo.GetAddress())
+			s.Require().NoError(err)
+
+			queryClient := types.NewQueryClient(clientCtx)
+			res, err := queryClient.QueryCellars(context.Background(), &types.QueryCellarsRequest{})
+			if err != nil {
+				return false
+			}
+			if res == nil {
+				return false
+			}
+			for _, c := range res.Cellars {
+				if c.Id == hardhatCellar.String() {
+					return true
+				}
+			}
+			return false
+		}, 30*time.Second, 2*time.Second, "hardhat cellar not found in chain")
+
+		// replace vote period check - sending commits with grpc call to steward
+		s.T().Logf("request rebalance start")
+		s.Require().Eventually(func() bool {
+			// Load in client TLS config
+			// client cert
+			clientCert, err := tls.LoadX509KeyPair(
+				"integration_tests/tls/client/test_client.crt",
+				"integration_tests/tls/client/test_client_key_pkcs8.pem",
+			)
+			s.Require().NoError(err)
+
+			// server CA
+			rootPool := x509.NewCertPool()
+			r, err := ioutil.ReadFile("integration_tests/tls/server/test_server_ca.crt")
+			s.Require().NoError(err)
+
+			block, _ := pem.Decode(r)
+			rootCert, err := x509.ParseCertificate(block.Bytes)
+			s.Require().NoError(err)
+
+			rootPool.AddCert(rootCert)
+
+			tlsConfig := &tls.Config{
+				Certificates: []tls.Certificate{clientCert},
+				RootCAs:      rootPool,
+				ServerName:   "",
+			}
+			creds := credentials.NewTLS(tlsConfig)
+
+			// make it rain
+			for i := range s.chain.stewards {
+				addr := fmt.Sprintf("localhost:%v", port+i)
+				conn, err := grpc.Dial(
+					addr,
+					grpc.WithBlock(),
+					grpc.WithTransportCredentials(creds),
+				)
+				s.Require().NoError(err)
+				defer conn.Close()
+
+				ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+				defer cancel()
+
+				c := NewUniswapV3CellarAllocatorClient(conn)
+				data := []*Position{
+					{
+						UpperPrice: commit.Vote.Cellar.TickRanges[0].Upper,
+						LowerPrice: commit.Vote.Cellar.TickRanges[0].Lower,
+						Weight:     commit.Vote.Cellar.TickRanges[0].Weight,
+					},
+				}
+				s.T().Logf("sending request to %s", s.chain.validators[i].keyInfo.GetAddress())
+				cellarId := fmt.Sprintf("%s:%s", "ethereum", hardhatCellar.String())
+				request := RebalanceRequest{CellarId: cellarId, Data: data}
+				_, err = c.Rebalance(ctx, &request)
+				s.Require().NoError(err)
+			}
+
+			return true
+		}, 100*time.Second, 1*time.Second, "rebalance request took too long")
+
```

Comment:
The tick ranges being eventually updated implies the success of the intermediate steps, but it would be good to add back test assertions to verify that the precommits and commits were successfully added to the chain. This is especially useful if one part of the overall flow breaks, otherwise it will be opaque to figure out what went wrong between data submission to steward and cellar update.

Reply:
Are debug logs in the steward containers enough? I tried for a little while to keep those tests in but they were having weird timing issues. I can add them back though.

Reply:
It's helpful to have tests fail in specific ways to indicate which bit is incorrect, and also to fail quicker in CI if they are going to break. Logs aren't very helpful for that bit.

Reply:
Done!